packages/Pagination/src/hooks/usePagination.ts

```diff
@@ -0,0 +1,91 @@
+import { useState, useEffect } from 'react';
+
+type PaginationHookProps = {
+  totalItems: number,
+  currentPage: number,
+  visiblePages: number,
+  defaultItemsPerPage?: number;
+  onPageChange: ({ pageNumber, itemsPerPage, offset }: { pageNumber: number, itemsPerPage: number, offset: number }) => void
+}
+
+type PaginationReturnType = {
+  page: number,
+  itemsPerPage: number,
+  calculatePages: () => number[],
+  handlePageChange: (pageNumber: number) => void,
+  handleOnItemsPerPage: (itemsNumber: number) => void,
+
+}
+
+const DEFAULT_ITEMS_PER_PAGE = 10;
+
+const usePagination = ({
+  totalItems,
+  currentPage,
+  visiblePages,
+  onPageChange,
+  defaultItemsPerPage
+}: PaginationHookProps): PaginationReturnType => {
+  const [page, setPage] = useState<number>(currentPage);
+  const [itemsPerPage, setItemsPerPage] = useState(DEFAULT_ITEMS_PER_PAGE);
+
+
+  useEffect(() => {
+    setPage(currentPage);
+  }, [currentPage]);
+
+  useEffect(() => {
+    defaultItemsPerPage && setItemsPerPage(defaultItemsPerPage);
+  }, [defaultItemsPerPage]);
+
+  const handlePageChange = (pageNumber: number) => {
+    setPage(pageNumber);
+    onPageChange({ pageNumber, itemsPerPage, offset: pageNumber * itemsPerPage });
+  };
+
+  const handleOnItemsPerPage = (itemsNumber: number) => {
+    setItemsPerPage(itemsNumber);
+    onPageChange({ pageNumber: page, itemsPerPage: itemsNumber, offset: (page - 1) * itemsPerPage });
+  }
+
+  const calculatePages = () => {
+    const pages: number[] = [];
+    const halfVisiblePages = Math.floor(visiblePages / 2);
+
+    if (totalItems <= visiblePages) {
```

Comment:
No need to add this change but here. isa cool way to initialize an array with n values

```
[...Array(n).keys()].map(i => i + 1)
```